### PR TITLE
Sol dir

### PR DESCRIPTION
**Pull request by ESeNonFossiIo**
this branch gives the possibility of:
- creating a directory for the solution
- creating a directory for every run of the code 
- copying prm file in this directory

### CODE REVIEW

**Comment by ESeNonFossiIo** [comment]
`/home/travis/dealii-inst/include/deal.II/bundled/boost/filesystem.hpp:15:12: fatal error: 'boost/filesystem/config.hpp' file not found`

In locale non ho questo errore e non so come risolverlo

**Comment by luca-heltai** [comment]
mmmm... this solution has the problem that requires a compiled version of boost (in fact, it fails on Travis) (it also fails if you DO have boost installed, but you do not export `BOOST_DIR`. Notice that the standard for boost is `BOOST_ROOT` and sometimes `BOOST_HOME`). 

I'd accept this patch if there was a mechanism to check for boost in the CMakeListst.txt, setting defining DEAL_II_SAK_WITH_BOOST, and the guarding the code that requires boost with #ifdef DEAL_II_SAK_WITH_BOOST, otherwise we wouldn't be able to use testing any more.

Moreover, the directory information is redundant. In principle you could specify as base name 

`outdir/base_file`

and the program would output everything in the directory `outdir`, or would bail out if the directory does not exist. Maybe we should not require people to have compiled boost installed... 

Can you see if you can write a small C++ only program that tests if a directory exists, and if not it creates it? something like `system("test -d`dirname " + base_name +"`|| mkdir`dirname " + base_name +"`);`

**Comment by ESeNonFossiIo** [comment]
I have added a precompiled flag to check if boost library is installed. Otherwise, it uses std::system. 

I prefer to not use std::system by default because it depends on the system we are using. On windows it would probably fails.

File: CMakeLists.txt [hunk 1]
```diff
@@ -34,6 +34,11 @@ SET(_d2_build_types "Release;Debug")
 SET(Release_postfix "")
 SET(Debug_postfix ".g")
 
+find_package(Boost)
+if(Boost_FOUND)
+  add_definitions(-DDEAL_II_SAK_WITH_BOOST)
+  SET(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} -I$ENV{BOOST_DIR}/include -L$ENV{BOOST_DIR}/lib -lboost_filesystem -lboost_system" )
```
Comment (luca-heltai) [inline CMakeLists.txt]: You should use the variables which are set by `find_package(boost);` this won't work unless you set `$BOOST_DIR`, which is not mandatory, even if you have installed boost, and cmake could find it...

File: tests/parsed_finite_element_04.cc [hunk 2]
```diff
@@ -19,9 +19,9 @@
 int main ()
 {
   initlog();
-  ParsedFiniteElement<2> fe_builder("FE", "FESystem[FE_Q(2)^d-FE_DGP(1)]", "u", 1);
+  ParsedFiniteElement<2> fe_builder("FE", "FESystem[FE_Q(2)^d-FE_DGP(1)]", "u,u,p", 1);
```
Comment (luca-heltai) [inline tests/parsed_finite_element_04.cc:22]: Why did you correct this? This test is checking that it actually fails when it is supposed to... (`stage=run`)...

Comment (ESeNonFossiIo) [inline tests/parsed_finite_element_04.cc:22]: because it failed to me. I revert my changes

**Pull request merged**